### PR TITLE
Allow empty lists in arguments

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
Passing empty lists/tuples to our arguments doesn't work

#### What happened in this PR?
- Allow empty lists and tuples in arguments
- Remove unnecessary CMake targets
- Add Pad operator tests

**JIRA TASK**: [DALI-XXXX]

### CODE REVIEW

File: dali/python/nvidia/dali/ops.py [hunk 1]
```diff
@@ -281,10 +281,11 @@ def __init__(self, **kwargs):
                   # as if the argument was not supplied at all
                   continue
 
-                if isinstance(value, list):
-                    if not value:
-                        raise RuntimeError("List arguments need to have at least 1 element.")
                 dtype = self._schema.GetArgumentType(key)
+                if isinstance(value, list) or isinstance(value, tuple):
```
Comment (Kh4L) [inline dali/python/nvidia/dali/ops.py]: Nitpicking
```suggestion
                if isinstance(value, (tuple, list)):
```

File: dali/pipeline/operator/argument.h [hunk 2]
```diff
@@ -205,9 +205,9 @@ class ArgumentInst<std::vector<T>> : public Argument {
 
   void SerializeToProtobuf(DaliProtoPriv* arg) override {
     const std::vector<T>& vec = val.Get();
-    DALI_ENFORCE(vec.size() > 0, "List arguments need to have at least 1 element.");
     arg->set_name(Argument::ToString());
-    arg->set_type(dali::serialize_type(vec[0]));
+    T dummy;
+    arg->set_type(dali::serialize_type(dummy));
```
Comment (mzient) [inline dali/pipeline/operator/argument.h]:
```suggestion
    arg->set_type(dali::serialize_type(T()));
```
?

File: dali/python/nvidia/dali/types.py [hunk 3]
```diff
@@ -69,6 +76,11 @@ def _type_convert_value(dtype, val):
         raise RuntimeError(str(dtype) + " does not correspond to a known type.")
     return _known_types[dtype][1](val)
 
+def _type_vector_element_type(dtype):
```
Comment (mzient) [inline dali/python/nvidia/dali/types.py]: Why not:
```suggestion
def _vector_element_type(dtype):
```
?

**Comment by jantonguirao** [comment]
!build

**Pull request merged**